src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -1067,15 +1067,13 @@ void LottieBuilder::updateText(LottieLayer* layer, float frameNo)
 
                     //text range process
                     for (auto s = text->ranges.begin(); s < text->ranges.end(); ++s) {
-                        float start, end;
-                        (*s)->range(frameNo, float(totalChars), start, end);
-
                         auto basedIdx = idx;
                         if ((*s)->based == LottieTextRange::Based::CharsExcludingSpaces) basedIdx = idx - space;
                         else if ((*s)->based == LottieTextRange::Based::Words) basedIdx = line + space;
                         else if ((*s)->based == LottieTextRange::Based::Lines) basedIdx = line;
 
-                        if (basedIdx < start || basedIdx >= end) continue;
+                        auto f = (*s)->factor(frameNo, float(totalChars), (float)basedIdx);
```

Comment:
1. Not sure the subsequent changes you mind now, probably, you can also consider use:
```
if ((*s)->factor(frameNo, float(totalChars), (float)basedIdx)) continue;
```
2. The name `factor` is too ambiguous in this context. I couldn't find any beneficial to change it from the original.

Reply:
please take a look here: https://github.com/thorvg/thorvg/pull/2898/files
I can push the other changes here if it's more readable.
this is a scaling factor that will be used to scale all text range styling properties. calling it a 'range' doesn't make a sense any more.

Reply:
@mgrudzinska If this patch alone doesn’t provide any benefit, please combine it with a patch that has a direct improvement or impact and submit them together in a single PR.